### PR TITLE
[#EVT-15] - All urls are secured by ApiKey (parameters)

### PR DESCRIPTION
**Pull request by marco-msg-ferrari**

### CODE REVIEW

File: src/EVT/ApiBundle/Resources/config/services.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+services:
+    evt_api_key_user_provider:
```
Comment (egulias) [inline src/EVT/ApiBundle/Resources/config/services.yml]: evt.api_key.user_provider ?

File: src/EVT/ApiBundle/Resources/config/services.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+        class: EVT\ApiBundle\Security\ApiKeyUserProvider
+        arguments: [%evt_api.apikeys%]
+
+    apikey_authenticator:
```
Comment (egulias) [inline src/EVT/ApiBundle/Resources/config/services.yml]: evt.api_key.authenticator ?

File: src/EVT/ApiBundle/Security/ApiKeyUserProvider.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        return new User($username, null, array('ROLE_API'));
+    }
+
+    public function refreshUser(UserInterface $user)
+    {
+        // this is used for storing authentication in the session
+        // but in this example, the token is sent in each request,
+        // so authentication can be stateless. Throwing this exception
+        // is proper to make things stateless
+        throw new UnsupportedUserException();
```
Comment (egulias) [inline src/EVT/ApiBundle/Security/ApiKeyUserProvider.php:44]: We can remove the comments :)

File: src/EVT/ApiBundle/Tests/Security/ApiKeyTest.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        $this->assertEquals('200', $statusCode);
+    }
+
+    public function testNoApikey()
+    {
+        $content = $this->client->request('GET', '/');
+        $statusCode = $this->client->getResponse()->getStatusCode();
+        $content = $this->client->getResponse()->getContent();
+
+        $this->assertEquals('401', $statusCode);
```
Comment (egulias) [inline src/EVT/ApiBundle/Tests/Security/ApiKeyTest.php:36]: `No api key` wouldn't be a 403 (forbidden)?

Comment (marco-msg-ferrari) [inline src/EVT/ApiBundle/Tests/Security/ApiKeyTest.php:36]: No, According to w3c:
"403 Forbidden

The server understood the request, but is refusing to fulfill it. Authorization will not help and the request SHOULD NOT be repeated...."

File: src/EVT/ApiBundle/Security/ApiKeyUserProvider.php (hunk 3 above)
Comment (marco-msg-ferrari) [inline src/EVT/ApiBundle/Security/ApiKeyUserProvider.php:44]: I think that this commet is usefull for understending why are we only throwing an exception withount condotions.

**Pull request merged**